src/Bicep.Core/Workspaces/SourceFileGroupingBuilder.cs

```diff
@@ -161,291 +149,135 @@ private ResultWithDiagnostic<ISourceFile> PopulateRecursive(Uri fileUri, Artifac
-            ArtifactReference? artifactReference,
-            Result<Uri, UriResolutionError> uriResult,
-            IFeatureProvider featureProvider,
-            RootConfiguration config,
-            BicepSourceFile file)
-        {
-            if (!featureProvider.ExtensibilityEnabled)
-            {
-                return new(x => x.ProvidersAreDisabled());
-            }
-
-            if (providerDeclarationSyntax.Specification is LegacyProviderSpecification { } legacySpecification)
-            {
-                if (!featureProvider.DynamicTypeLoadingEnabled && !featureProvider.ProviderRegistryEnabled)
-                {
-                    return new(ResourceTypesProviderDescriptor.CreateBuiltInProviderDescriptor(
-                        legacySpecification.NamespaceIdentifier,
-                        legacySpecification.Version,
-                        file.FileUri,
-                        providerDeclarationSyntax.Alias?.IdentifierName));
-                }
-                return new(x => x.ExpectedProviderSpecification());
-            }
-
-            if (!featureProvider.DynamicTypeLoadingEnabled && !featureProvider.ProviderRegistryEnabled)
-            {
-                return new(x => x.UnrecognizedProvider(providerDeclarationSyntax.Specification.NamespaceIdentifier));
-            }
-
-            if (providerDeclarationSyntax.Specification is InlinedProviderSpecification { } inlinedSpecification)
-            {
-                if (!uriResult.IsSuccess(out var typesTgzUri, out var uriResolutionError))
-                {
-                    return new(uriResolutionError.ErrorBuilder);
-                }
-                return new(ResourceTypesProviderDescriptor.CreateDynamicallyLoadedProviderDescriptor(
-                    inlinedSpecification.NamespaceIdentifier,
-                    inlinedSpecification.Version,
-                    file.FileUri,
-                    artifactReference,
-                    typesTgzUri,
-                    providerDeclarationSyntax.Alias?.IdentifierName));
-            }
-
-            // The provider is configuration managed, fetch the provider source & version from the configuration
-            var configLookupValue = providerDeclarationSyntax.Specification.NamespaceIdentifier;
-
-            // Special case the "sys" provider for backwards compatibility
-            if (configLookupValue == SystemNamespaceType.BuiltInName)
-            {
-                return new(ResourceTypesProviderDescriptor.CreateBuiltInProviderDescriptor(
-                    SystemNamespaceType.BuiltInName,
-                    ResourceTypesProviderDescriptor.LegacyVersionPlaceholder,
-                    file.FileUri));
-            }
-            return TryGetDescriptorForConfigManagedProvider(configLookupValue, file, config);
-        }
-
-        private void ProcessImplicitProviderDeclarations(BicepSourceFile bicepFile, ProviderDescriptorBundleBuilder bundleBuilder, RootConfiguration rootConfig)
```

Comment:
This is where declarations of implicitly imported providers is processed, it used to add to the bundle those providers that have no syntax node associated to them.